aggregator/src/aggregator.rs

```diff
@@ -3182,6 +3292,80 @@ mod tests {
         });
     }
 
+    #[tokio::test]
+    async fn upload_report_encrypted_with_global_key() {
+        install_test_trace_subscriber();
+
+        let (vdaf, aggregator, clock, task, datastore, _ephemeral_datastore) =
+            setup_upload_test(Config {
+                max_upload_batch_size: 1000,
+                max_upload_batch_write_delay: StdDuration::from_millis(500),
+                global_hpke_configs_refresh_interval: StdDuration::from_millis(500),
+                ..Default::default()
+            })
+            .await;
+
+        // Same ID as the task to test having both keys to choose from.
+        let global_hpke_keypair_same_id = generate_test_hpke_config_and_private_key_with_id(
+            (*task.current_hpke_key().config().id()).into(),
+        );
+        // Different ID to test misses on the task key.
+        let global_hpke_keypair_different_id = generate_test_hpke_config_and_private_key_with_id(
+            (0..)
+                .map(HpkeConfigId::from)
+                .find(|id| !task.hpke_keys().contains_key(id))
+                .unwrap()
+                .into(),
+        );
+
+        datastore
+            .run_tx(|tx| {
+                let global_hpke_keypair_same_id = global_hpke_keypair_same_id.clone();
+                let global_hpke_keypair_different_id = global_hpke_keypair_different_id.clone();
+                Box::pin(async move {
+                    // Leave these in the PENDING state--they should still be decryptable.
+                    tx.put_global_hpke_keypair(&global_hpke_keypair_same_id)
+                        .await?;
+                    tx.put_global_hpke_keypair(&global_hpke_keypair_different_id)
+                        .await?;
+                    Ok(())
+                })
+            })
+            .await
+            .unwrap();
+
+        // Let keypair cache refresh.
+        sleep(StdDuration::from_millis(750)).await;
```

Comment:
We should avoid this kind of thing in tests, as they tend to be brittle and increase test runtime. Could you instead expose some test method on the keypair cache or on the aggregator struct that lets you force the cache refresh?

Reply:
Agreed, it does occur to me that if the build machine is overloaded (which is likely, they're fairly weak), the cache may not refresh in time.

Can I make the change in a separate PR? I'll have to modify existing tests.

Reply:
Here's how that'll look roughly https://github.com/divviup/janus/commit/655cc8a0736b293e706e953c019b78e5c8b0e438 (I should have done this in the first place!)

Reply:
Sure, we can take https://github.com/divviup/janus/commit/655cc8a0736b293e706e953c019b78e5c8b0e438 separately so long as we don't let it linger.